python-checks/src/main/java/org/sonar/python/checks/DeprecatedNumpyTypesCheck.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks;
+
+import java.util.Map;
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.quickfix.PythonQuickFix;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.QualifiedExpression;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.python.quickfix.TextEditUtils;
+
+@Rule(key = "S6730")
+public class DeprecatedNumpyTypesCheck extends PythonSubscriptionCheck {
+
+  private static final String MESSAGE = "Replace this deprecated numpy type alias with the builtin type %s.";
+  private static final String QUICK_FIX_MESSAGE = "Replace with %s.";
+
+  private static final Map<String, String> typesToCheck = Map.of(
+      "numpy.bool", "bool",
+      "numpy.int", "int",
+      "numpy.float", "float",
+      "numpy.complex", "complex",
+      "numpy.object", "object",
+      "numpy.str", "str",
+      "numpy.long", "int",
+      "numpy.unicode", "str");
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.QUALIFIED_EXPR,
+        DeprecatedNumpyTypesCheck::checkForDeprecatedTypesNames);
+  }
+
+  private static void checkForDeprecatedTypesNames(SubscriptionContext ctx) {
+    QualifiedExpression expression = (QualifiedExpression) ctx.syntaxNode();
+    Symbol symbol = expression.symbol();
+    checkForNumpyType(symbol, expression, ctx);
+  }
+
+  private static void checkForNumpyType(@Nullable Symbol symbol, Tree expression, SubscriptionContext ctx) {
```

Comment:
Minor: I think there is not that much sense to have the separate method here. I guess the whole logic could be done in the `checkForDeprecatedTypeNames` method like:
```java
Optional.of(ctx.syntaxNode)
  .map(QualifiedExpression.class::cast)
  .ifPresent(expression -> Optional.ofNullable(expression.symbol)
    .map(Symbol::fullyQualifiedName)
    .map(typesToCheck::get)
    .ifPresent(type -> raiseIssue(expression, type, ctx)));
```

Reply:
Yes, good catch! Thank you. This was a left over from some old implementation.